db/db_impl/db_impl_compaction_flush.cc

```diff
@@ -1551,13 +1551,22 @@ Status DBImpl::FlushMemTable(ColumnFamilyData* cfd,
     if (!cfd->mem()->IsEmpty() || !cached_recoverable_state_empty_.load()) {
       s = SwitchMemtable(cfd, &context);
     }
-
     if (s.ok()) {
       if (cfd->imm()->NumNotFlushed() != 0 || !cfd->mem()->IsEmpty() ||
           !cached_recoverable_state_empty_.load()) {
         flush_memtable_id = cfd->imm()->GetLatestMemTableID();
         flush_req.emplace_back(cfd, flush_memtable_id);
       }
+      if (immutable_db_options_.persist_stats_to_disk) {
+        ColumnFamilyData* cfd_stats =
+            versions_->GetColumnFamilySet()->GetColumnFamily(
+                kPersistentStatsColumnFamilyName);
+        if (cfd_stats != cfd && cfd_stats != nullptr) {
+          s = SwitchMemtable(cfd_stats, &context);
```

Comment:
To confirm, do we allow stats cf to be dropped?

Reply:
adding code do disallow dropping stats cf

Reply:
I think not exposing the ColumnFamilyHandle of the stats cf will suffice.

Reply:
PersistentStatsColumnFamily() is not part of public interface so I guess it's fine?

Reply:
SG